client/test/e2e/patient_invoice/modal.page.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Created by Dedrick Kitamuka on 15/06/2016.
+ */
+
+function modalPage () {
+    var page = this;
+
+    var todayButton = element(by.id('today'));
+    var weekButton = element(by.id('week'));
+    var monthButton = element(by.id('month'));
+    var yearButton = element(by.id('year'));
+
+    var referenceField = element(by.id('reference'));
+
+    var serviceSelect = element(by.id('service'));
+    var userSelect = element(by.id('user'));
+
+    var allRadio = element(by.id('all'));
+    var yesRadio = element(by.id('yes'));
+    var noRadio = element(by.id('no'));
+
+
+    var submitButton = element(by.id('submitButton'));
+
+
+    //interfaces
+    function chooseToday() {
```

Comment:
This code is fine - to reduce the number of lines in this page, it might be easier to make more generic functions that use parameters.  For example:
```
// this will do chooseToday()
modal.range('today');

// this will do chooseWeek
modal.range('week');

// this will do chooseMonth
modal.range('month');

// ... etc ..
```
That way there is less code to maintain going forwards.  These tests are excellent though.